File: Coding Ninjs/Medium/Count Of 3s.py
def countOf3( x ):
    s = str( x )
    m = len( s ) - 1
    a = [ 0 ] * ( m + 1 )
    ans = 1 if x % 10 >= 3 else 0
    
    for i in range( m ):
        a[ i + 1 ] = a[i] * 10 + 10 ** i
        
    for i in range( m ):
        if s[i] > "3":
            ans += 10 * a[ m - i ]
        elif s[i] == "3":
            ans += x % 10 ** ( m - i ) + 1
            
        ans += int( s[i] ) * a[ m - i ]
        
    return ans

File: Coding Ninjas/Moderate/Count Of 3s.java
public class Solution
{
    public static long countOf3
    ( int x )
    {
        String s = Integer.toString( x );
        int m = s.length() - 1;
        int[] a = new int[ m + 1 ];
        long ans = 0;

        for ( int i = 0; i < m; i++ )
        {
            a[ i + 1 ] = a[i] * 10 + ( int ) Math.pow( 10, i );
        }  
        
        for ( int i = 0; i <= m; i++ )
        {
            if ( s.charAt( i ) > '3' )
            {
                ans += ( int ) Math.pow( 10, m - i );
            }
            else if ( s.charAt( i ) == '3' )
            {
                ans += ( x % ( int ) Math.pow( 10, m - i ) ) + 1;
            }
            
            ans += a[ m - i ] * Character.getNumericValue( s.charAt( i ) );
        }

        return ans;
    }

}
